front/app/hooks/usePages.ts

```diff
@@ -39,17 +40,31 @@ function createSubscription(inputProps$, setPages) {
 
           return combineLatest(
             ids.map((id) =>
-              pageByIdStream(id).observable.pipe(map((topic) => topic.data))
+              pageByIdStream(id).observable.pipe(
+                map((response) => {
+                  return isNilOrError(response) ? response : response.data;
+                })
+              )
             )
           );
         }
 
         return listPages().observable.pipe(
-          map((pages) => {
-            return pages.data;
+          map((response) => {
+            return isNilOrError(response) ? response : response.data;
           })
         );
       })
     )
-    .subscribe(setPages);
+    .subscribe((pages) => {
+      if (isNilOrError(pages)) {
+        setPages(pages);
+        return;
+      }
+
+      const nilOrErrorPages = pages.filter(isNilOrError);
+      nilOrErrorPages.length > 0
+        ? setPages(nilOrErrorPages[0])
```

Comment:
So will this set all pages to a page that's nil or error?

Reply:
Yeah, the idea is that if you provide an array of ideas, it makes a request to the `static_pages/:id` endpoint for each id. If any of those fails, we set `pages` to nil or error. Not an ideal solution but better than having no error handling at all, which is how it was before...